### DISCUSSION TITLE
xwayland-satellite - rootless xwayland for niri

### DISCUSSION THREAD

**Discussion by Supreeeme** [YaLTeR/niri#318] [Show and tell]
Here's something I've been working on for the last few months: https://github.com/Supreeeme/xwayland-satellite

xwayland-satellite is rootless xwayland that lives outside of the compositor. I made it for niri, but it should work with any compositor implementing `xdg_wm_base`. I primarily made it for Steam, so it's likely missing a lot of features and might even crash from time to time, so feel free to open some bugs and I'll work on fixing them!

**Comment by nebulosa2007** [comment]
Could you add a tag to the last commit or make a release? I want to try your implementation, but don't want to make another `-git` package...

**Reply by Supreeeme** [reply]
Tagged the last commit!

**Comment by Okanda** [comment]
Hello. I don't understand how to use this in niri.

**Reply by Supreeeme** [reply]
You can just run it while in niri (with `cargo run`). It then prints out what display it's connected to, so then you launch any X11 apps with `DISPLAY=<display>` set (where `<display>` is probably `:0`). You can also set it to run at startup in niri and then add the display to the environment section of the config.

**Comment by Okanda** [comment]
So simple ! Thank you :)

**Comment by Dimev** [comment]
Very nice, thanks!
